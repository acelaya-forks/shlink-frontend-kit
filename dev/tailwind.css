@import 'tailwindcss' prefix(tw) important;
@import '../src/tailwind.preset.css';
