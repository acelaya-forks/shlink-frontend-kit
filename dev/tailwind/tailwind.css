@import 'tailwindcss' prefix(tw) important;
@import '../../src/tailwind/tailwind.preset.css';
