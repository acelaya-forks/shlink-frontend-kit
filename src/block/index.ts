export * from './Message';
export * from './Result';
export * from './SimpleCard';
