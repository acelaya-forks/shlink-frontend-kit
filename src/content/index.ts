export * from './Details';
export * from './Listbox';
export * from './Table';
