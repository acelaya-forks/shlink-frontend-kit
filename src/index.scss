@import './utils/ResponsiveTable';
@import './theme/theme';
