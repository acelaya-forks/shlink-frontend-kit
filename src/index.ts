export * from './block';
export * from './form';
export * from './hooks';
export * from './navigation';
export * from './ordering';
