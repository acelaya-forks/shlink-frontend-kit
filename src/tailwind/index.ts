export * from './content';
export * from './feedback';
export * from './form';
export * from './helpers';
export * from './navigation';
export * from './surfaces';
export * from './utils';
export * from './types';
