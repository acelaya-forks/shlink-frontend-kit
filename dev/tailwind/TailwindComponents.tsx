import type { FC } from 'react';
import { Route, Routes } from 'react-router';
import { DetailsPage } from './content/DetailsPage';
import { TablePage } from './content/TablePage';
import { MessagePage } from './feedback/MessagePage';
import { ModalDialogPage } from './feedback/ModalDialogPage';
import { ResultPage } from './feedback/ResultPage';
import { TooltipPage } from './feedback/TooltipPage';
import { ButtonsPage } from './form/ButtonsPage';
import { InputsPage } from './form/InputsPage';
import { SearchComboboxPage } from './form/SearchComboboxPage';
import { SearchInputPage } from './form/SearchInputPage';
import { NavPillsPage } from './navigation/NavPillsPage';
import { PaginatorPage } from './navigation/PaginatorPage';
import { CardsPage } from './surfaces/CardsPage';
import { UtilsPage } from './utils/UtilsPage';

export const TailwindComponents: FC = () => {
  return (
    <Routes>
      <Route path="/" element={<h2 className="text-center">Select component - Tailwind</h2>} />
      <Route path="/form/inputs" element={<InputsPage />} />
      <Route path="/form/search-input" element={<SearchInputPage />} />
      <Route path="/form/search-combobox" element={<SearchComboboxPage />} />
      <Route path="/form/buttons" element={<ButtonsPage />} />
      <Route path="/surfaces/cards" element={<CardsPage />} />
      <Route path="/content/tables" element={<TablePage />} />
      <Route path="/content/details" element={<DetailsPage />} />
      <Route path="/navigation/paginator" element={<PaginatorPage />} />
      <Route path="/navigation/nav-pills">
        <Route path="" element={<NavPillsPage />} />
        <Route path="*" element={<NavPillsPage />} />
      </Route>
      <Route path="/feedback/dialogs" element={<ModalDialogPage />} />
      <Route path="/feedback/result" element={<ResultPage />} />
      <Route path="/feedback/message" element={<MessagePage />} />
      <Route path="/feedback/tooltip" element={<TooltipPage />} />
      <Route path="/utils" element={<UtilsPage />} />
      <Route path="*" element={<h2 className="text-center">Not found - Tailwind</h2>} />
    </Routes>
  );
};
