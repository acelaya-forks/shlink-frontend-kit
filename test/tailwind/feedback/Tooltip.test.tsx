import { screen, waitFor, waitForElementToBeRemoved } from '@testing-library/react';
import type { UseTooltipOptions } from '../../../src/tailwind';
import { Tooltip, useTooltip } from '../../../src/tailwind';
import { checkAccessibility } from '../../__helpers__/accessibility';
import { renderWithEvents } from '../../__helpers__/setUpTest';

function TestComponent(options: UseTooltipOptions) {
  const { anchor, tooltip } = useTooltip(options);

  return (
    <div>
      <div data-testid="anchor" {...anchor}>Anchor</div>
      <Tooltip {...tooltip}>Hello!</Tooltip>
    </div>
  );
}

describe('<Tooltip />', () => {
  const setUp = (props: UseTooltipOptions = {}) => renderWithEvents(<TestComponent {...props} />);

  it('passes a11y checks', async () => {
    const { user } = setUp();
    await user.hover(screen.getByTestId('anchor'));
    await screen.findByRole('tooltip');

    return checkAccessibility(setUp());
  });

  it('renders tooltip on hover with transition', async () => {
    const { user } = setUp();

    expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
    await user.hover(screen.getByTestId('anchor'));
    await waitFor(() => expect(screen.getByRole('tooltip')).toBeInTheDocument());

    await user.unhover(screen.getByTestId('anchor'));
    await waitForElementToBeRemoved(screen.getByRole('tooltip'));
  });

  it.each([
    'top' as const,
    'bottom' as const,
    'left' as const,
    'right' as const,
  ])('renders arrow in the proper location based on placement option', async (placement) => {
    const { user } = setUp({ placement });

    await user.hover(screen.getByTestId('anchor'));
    const tooltip = await screen.findByRole('tooltip');
    const arrow = screen.getByTestId('arrow');

    expect(`${tooltip.className}_${arrow.className}`).toMatchSnapshot();
  });
});
