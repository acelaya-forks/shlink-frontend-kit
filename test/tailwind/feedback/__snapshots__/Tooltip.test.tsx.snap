// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Tooltip /> > renders arrow in the proper location based on placement option 1`] = `"tw:z-500 tw:max-w-64 tw:pb-2.5_tw:absolute tw:border-l-6 tw:border-r-6 tw:border-b-6 tw:border-l-transparent tw:border-r-transparent tw:border-b-black/90 tw:rotate-180"`;

exports[`<Tooltip /> > renders arrow in the proper location based on placement option 2`] = `"tw:z-500 tw:max-w-64 tw:pt-2.5_tw:absolute tw:border-l-6 tw:border-r-6 tw:border-b-6 tw:border-l-transparent tw:border-r-transparent tw:border-b-black/90"`;

exports[`<Tooltip /> > renders arrow in the proper location based on placement option 3`] = `"tw:z-500 tw:max-w-64 tw:pr-2.5_tw:absolute tw:border-l-6 tw:border-r-6 tw:border-b-6 tw:border-l-transparent tw:border-r-transparent tw:border-b-black/90 tw:rotate-90 tw:mr-[-3px]"`;

exports[`<Tooltip /> > renders arrow in the proper location based on placement option 4`] = `"tw:z-500 tw:max-w-64 tw:pl-2.5_tw:absolute tw:border-l-6 tw:border-r-6 tw:border-b-6 tw:border-l-transparent tw:border-r-transparent tw:border-b-black/90 tw:rotate-270 tw:ml-[-3px]"`;
